providers/core/core.rtcdatachannel.js

```diff
@@ -124,9 +125,16 @@ RTCDataChannelAdapter.prototype.onclose = function (event) {
 RTCDataChannelAdapter.prototype.onmessage = function (event) {
   if (typeof event.data === 'string') {
     this.dispatchEvent('onmessage', {text: event.data});
-  } else {
+  } else if (event.data instanceof ArrayBuffer) {
     this.dispatchEvent('onmessage', {buffer: event.data});
+  } else if (event.data instanceof Blob) {
```

Comment:
Very good point although if Freedom can't support `Blob` then perhaps it should always convert to an `ArrayBuffer`?